tests/test_ssm/test_ssm_boto3.py

```diff
@@ -48,6 +48,44 @@ def test_delete_parameters():
 
 
 @mock_ssm
+def test_get_parameters_by_path():
+    client = boto3.client('ssm', region_name='us-east-1')
+
+    client.put_parameter(
+        Name='/foo/name1',
+        Description='A test parameter',
+        Value='value1',
+        Type='String')
+
+    client.put_parameter(
+        Name='/foo/name2',
+        Description='A test parameter',
+        Value='value2',
+        Type='String')
+
+    client.put_parameter(
+        Name='/bar/name3',
+        Description='A test parameter',
+        Value='value3',
+        Type='String')
+
+    client.put_parameter(
+        Name='/bar/name3/name4',
+        Description='A test parameter',
+        Value='value3',
+        Type='String')
+
+    response = client.get_parameters_by_path(Path='/foo')
+    len(response['Parameters']).should.equal(2)
+
+    response = client.get_parameters_by_path(Path='/bar', Recursive=False)
+    len(response['Parameters']).should.equal(1)
+
+    response = client.get_parameters_by_path(Path='/bar', Recursive=True)
+    len(response['Parameters']).should.equal(2)
```

Comment:
It would be helpful to add tests for the response content, too. Something like:

```
response = client.get_parameters_by_path(Path='/bar', Recursive=True)
len(response['Parameters']).should.equal(2)
set([p['Value'] for p in response['Parameters']).should.equal(set(['value3', 'value4']))
```